invenio_communities/assets/semantic-ui/js/invenio_communities/members/members/components/MembersResult.js

```diff
@@ -8,51 +8,80 @@
 
 import React from "react";
 import { Grid } from "semantic-ui-react";
-import { ResultsList, Pagination, ResultsPerPage, Count } from "react-searchkit";
+import {
+  ResultsList,
+  Pagination,
+  ResultsPerPage,
+  Count,
+} from "react-searchkit";
 import { i18next } from "@translations/invenio_communities/i18next";
+import { ModalManager } from "../../components/modal_manager";
+import {
+  LeaveCommunityModal,
+  RemoveMemberModal,
+} from "../../components/Modals";
+
+export const modalEnum = {
+  leaveCommunity: 1,
+  removeMember: 2,
+};
+
+const modalMap = {
+  [modalEnum.leaveCommunity]: LeaveCommunityModal,
+  [modalEnum.removeMember]: RemoveMemberModal,
+};
 
 export const MembersResults = ({ paginationOptions, currentResultsState }) => {
   const { total } = currentResultsState.data;
+
   return (
     total && (
-      <Grid>
-        <Grid.Row>
-          <Grid.Column width={16}>
-            <Grid>
-              <Grid.Row>
-                <Grid.Column>
-                  <ResultsList />
-                </Grid.Column>
-              </Grid.Row>
-            </Grid>
-          </Grid.Column>
-        </Grid.Row>
-        <Grid.Row verticalAlign="middle">
-          <Grid.Column width={4}>
-            <Count label={() => <>{total} {i18next.t("result(s) found")}</>} />
-          </Grid.Column>
-          <Grid.Column width={8} textAlign="center">
-            <Pagination
-              options={{
-                size: "mini",
-                showFirst: false,
-                showLast: false,
-              }}
-            />
-          </Grid.Column>
-          <Grid.Column textAlign="right" width={4}>
-            <ResultsPerPage
-              values={paginationOptions.resultsPerPage}
-              label={(cmp) => (
-                <>
-                  {" "}
-                  {cmp} {i18next.t("results per page")}
-                </>
-              )}
-            />
-          </Grid.Column>
-        </Grid.Row>
-      </Grid>
+      <ModalManager modalMap={modalMap}>
```

Comment:
do you need to wrap the whole list into context or only `<ResultsList>`?

Reply:
it should actually only be wrapping the `ResultsList`, good one.